src/plugins/intel_gpu/src/graph/impls/ocl/kernel_selector_helper.h

```diff
@@ -233,6 +233,14 @@ inline ov::PartialShape extend_shape_to_rank_from_begin(const ov::PartialShape&
     return extended_pshape;
 }
 
+inline ov::PartialShape extend_shape_to_rank_from_idx_equal_to_diff(ov::PartialShape pshape, size_t rank = 4) {
+    if (pshape.size() >= rank) {
+        return pshape;
+    }
+    pshape.insert(pshape.begin() + (rank - pshape.size()), ov::Dimension(1));
+    return pshape;
+}
+
```

Comment:
this function is difficult to understand the intention. If this is not a general function, I think it would be better to place it in the target cpp file, with better name..